overlord/devicestate/devicemgr.go

```diff
@@ -782,6 +782,20 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// canAutoRefresh is a helper that checks if the device is able to
+// auto-refresh
+func canAutoRefresh(st *state.State) bool {
+	// no need to wait for seeding on classic
+	if release.OnClassic {
+		return true
+	}
+
+	// on all-snap devices we need to be seeded first
+	var seeded bool
+	st.Get("seeded", &seeded)
+	return seeded
+}
```

Comment:
I suggest starting on the safe side: block until registration is done. Otherwise we'll need to think through the consequences of running those very early initialization procedures concurrently with arbitrary refreshes, and the running of arbitrary code inside the relevant hooks.